transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollServerChannel.java

```diff
@@ -75,11 +76,19 @@ public void connect(SocketAddress socketAddress, SocketAddress socketAddress2, C
         @Override
         void epollInReady() {
             assert eventLoop().inEventLoop();
+            boolean edgeTriggered = isFlagSet(Native.EPOLLET);
+
+            final ChannelConfig config = config();
+            if (!readPending && !config.isAutoRead() && !edgeTriggered) {
```

Comment:
nit: Consider removing `config` variable and just calling `config()`.  It is only used 1 time, and with short circuit evaluation you may never have to call...you could event move it further down the `&&` chain if you wanted.